### PR TITLE
Add Student ID when student role is selected

### PR DESCRIPTION
**Pull request by nickscialli-usds**
## Related Issue or Background Info

- #1170

## Changes Proposed

- Displays Student ID when the `"STUDENT"` role is selected
- Uses the existing `lookupId` field on Person

## Screenshot

student ID appearing after role

### CODE REVIEW

File: frontend/e2e/pages/patients.js [hunk 1]
```diff
@@ -29,6 +31,9 @@ function addPatient(dobFormat) {
   this.section.editPatient.setValue("@address", address);
   this.section.editPatient.setValue("@state", state);
   this.section.editPatient.setValue("@zip", zip);
+  this.section.editPatient.setValue("@role", role);
+  this.expect.section("@editPatient").to.contain.text("Student ID");
```
Comment (nickscialli-usds) [inline frontend/e2e/pages/patients.js:35]: Oddly this seems to fail during CI but passes locally (using Chrome). Does anyone know what's going on here? The field _should_ dynamically appear after the role is set to `"STUDENT"`. I'm assuming `this.expect` has some retry interval and this is not a timing thing but I'm not sure.

**Comment by nickscialli-usds** [comment]
Realizing that I didn't really address pxp in this PR. Do we know if patients should be able to update their Student ID? Should they be able to edit their role?

**Comment by katiealoisi** [comment]
Looks good to me!

File: frontend/src/app/patients/Components/PersonForm.tsx [hunk 2]
```diff
@@ -209,6 +208,9 @@ const PersonForm = (props: Props) => {
             options={ROLE_VALUES}
             defaultSelect={true}
           />
+          {patient.role === "STUDENT" && (
```
Comment (nullflux) [inline frontend/src/app/patients/Components/PersonForm.tsx:211]: Concern only tangentially related to this PR: in `constants.d.ts`, the `Role` type is a union of several string constants. There also exists, in `constants/index.tsx`, `ROLE_VALUES`, which enumerates some label/value pairs for this same type.

I wonder if the `Role` type should be, instead, a string-valued Enum which then keys the VALUES enum? 

But this refactor is beyond the scope of this magic string constant, which I hope, would be deprecated if/when these display values become configurable in the future

**Review by nullflux — APPROVED** [review]
Approved with an inlined and not-important comment about constants that requires no action! ✅

**Review by ClydeDroid — APPROVED** [review]

**Pull request merged**